app/presenters/two_factor_login_options_presenter.rb

```diff
@@ -140,4 +143,15 @@ def sp_name
       APP_NAME
     end
   end
+
+  def confirmation_period
```

Comment:
The use of `ConfirmationEmailPresenter` in other controllers like `account_reset/request_controller.rb` felt a little unexpected, but I understood it to be based on wanting to reuse the logic of `confirmation_period`.  Here we're we're reimplementing it. I'm guessing the challenge was that we don't have easy access to the constructor arguments required for `ConfirmationEmailPresenter` ?

To maintain consistency, I'd think we should either...

- Use `ConfirmationEmailPresenter#confirmation_period` consistently
- Reimplement consistently in all uses
- Move `confirmation_period` and/or create a dedicated service class for that common logic

Reply:
For Option#2, maybe we can simplify the implementation so it's a little less bulky to copy everywhere, e.g.

```rb
distance_of_time(Devise.confirm_within, accumulate_on: :hours)
```

Ref: https://github.com/radar/distance_of_time_in_words?tab=readme-ov-file#distance_of_time

Reply:
My understanding was that there are instances where `confirmation_period` in `ConfirmationEmailPresenter` could differ from the `account_reset_wait_period_days` period because one has the confirmation period in `Devise` as opposed to the other which can be set by whoever wants to change it. If what I am understanding is the case, then we do want to re-use the logic, but with different variables.

Reply:
> My understanding was that there are instances where `confirmation_period` in `ConfirmationEmailPresenter` could differ from the `account_reset_wait_period_days` because one has the confirmation period in `Devise` as opposed to the other which can be set by whoever wants to change it. If what I am understanding is the case, then we do want to re-use the logic, but with different variables.

Oh! I hadn't realized the difference between `Devise.confirm_within` for new email confirmation vs. `IdentityConfig.store.account_reset_wait_period_days` for account reset delay. I think them being the same timeframe confused me, though we definitely should consider them as different things.